src/panel/direction/DirectionPanel.jsx

```diff
@@ -16,6 +16,15 @@ import { fire, listen, unListen } from 'src/libs/customEvents';
 import * as address from '../../libs/address';
 import { CloseButton, Flex } from '../../components/ui';
 
+const setMarker = (type, poi) => {
+  window.execOnMapLoaded(() => {
+    fire(`set_${type}`, poi);
```

Comment:
I am not very fond of using a dynamic value here. It would make future refactoring particularly harder if we can't even `grep` for the event name. Maybe we could merge `set_destination` and `set_origin` into a single event by adding another parameter.